<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>rncbc aka Rui Nuno Capela</author>
 <comment>qtractor - An Audio/MIDI multi-track sequencer.

   Copyright (C) 2005-2024, rncbc aka Rui Nuno Capela. All rights reserved.

   This program is free software; you can redistribute it and/or
   modify it under the terms of the GNU General Public License
   as published by the Free Software Foundation; either version 2
   of the License, or (at your option) any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License along
   with this program; if not, write to the Free Software Foundation, Inc.,
   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

 </comment>
 <class>qtractorClipForm</class>
 <widget class="QDialog" name="qtractorClipForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>420</width>
    <height>540</height>
   </rect>
  </property>
  <property name="focusPolicy">
   <enum>Qt::StrongFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Clip</string>
  </property>
  <property name="windowIcon">
   <iconset resource="qtractor.qrc">:/images/editClip.png</iconset>
  </property>
  <layout class="QGridLayout">
   <item row="0" column="0" colspan="10">
    <widget class="QLabel" name="ClipNameTextLabel">
     <property name="text">
      <string>&amp;Name:</string>
     </property>
     <property name="buddy">
      <cstring>ClipNameLineEdit</cstring>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="10">
    <widget class="QLineEdit" name="ClipNameLineEdit">
     <property name="toolTip">
      <string>Clip name</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="10">
    <widget class="QLabel" name="FilenameTextLabel">
     <property name="text">
      <string>&amp;File:</string>
     </property>
     <property name="buddy">
      <cstring>FilenameComboBox</cstring>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="9">
    <widget class="QComboBox" name="FilenameComboBox">
     <property name="toolTip">
      <string>Clip filename</string>
     </property>
     <property name="editable">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="3" column="9">
     <widget class="QToolButton" name="FilenameToolButton">
     <property name="minimumSize">
      <size>
       <width>22</width>
       <height>22</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>22</width>
       <height>22</height>
      </size>
     </property>
     <property name="focusPolicy">
      <enum>Qt::TabFocus</enum>
     </property>
     <property name="toolTip">
      <string>Browse for clip file</string>
     </property>
     <property name="icon">
      <iconset resource="qtractor.qrc">:/images/fileOpen.png</iconset>
     </property>
    </widget>
   </item>
   <item row="4" column="0" colspan="7">
    <spacer>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>8</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="4" column="7">
    <widget class="QLabel" name="TrackChannelTextLabel">
     <property name="text">
      <string>Track/&amp;Channel:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignVCenter</set>
     </property>
     <property name="buddy">
      <cstring>TrackChannelSpinBox</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="8" colspan="2">
    <widget class="QSpinBox" name="TrackChannelSpinBox">
     <property name="minimumSize">
      <size>
       <width>40</width>
       <height>0</height>
      </size>
     </property>
     <property name="toolTip">
      <string>Clip track/channel</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0" colspan="10">
    <widget class="QGroupBox" name="ParamGroupBox">
     <property name="title">
      <string>Parameters</string>
     </property>
     <layout class="QGridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="ClipStartTextLabel">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>&amp;Start:</string>
        </property>
        <property name="buddy">
         <cstring>ClipStartSpinBox</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qtractorTimeSpinBox" name="ClipStartSpinBox">
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Clip start</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2" rowspan="3">
       <spacer>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="3" colspan="2">
       <widget class="QLabel" name="ClipGainTextLabel">
        <property name="text">
         <string>&amp;Gain:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>ClipGainSpinBox</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QDoubleSpinBox" name="ClipGainSpinBox">
        <property name="minimumSize">
         <size>
          <width>66</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Clip gain/volume</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="accelerated">
         <bool>true</bool>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="singleStep">
         <double>0.1</double>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="ClipOffsetTextLabel">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Offs&amp;et:</string>
        </property>
        <property name="buddy">
         <cstring>ClipOffsetSpinBox</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qtractorTimeSpinBox" name="ClipOffsetSpinBox">
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Clip offset</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3" colspan="2">
       <widget class="QLabel" name="ClipPanningTextLabel">
        <property name="text">
         <string>&amp;Panning:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>ClipPanningSpinBox</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="5">
       <widget class="QDoubleSpinBox" name="ClipPanningSpinBox">
        <property name="minimumSize">
         <size>
          <width>66</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Clip gain/volume</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="accelerated">
         <bool>false</bool>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="singleStep">
         <double>0.1</double>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="ClipLengthTextLabel">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>&amp;Length:</string>
        </property>
        <property name="buddy">
         <cstring>ClipLengthSpinBox</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="qtractorTimeSpinBox" name="ClipLengthSpinBox">
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Clip length</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QLabel" name="FormatTextLabel">
        <property name="text">
         <string>Forma&amp;t:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>FormatComboBox</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="4" colspan="2">
       <widget class="QComboBox" name="FormatComboBox">
        <property name="toolTip">
         <string>Time display format</string>
        </property>
        <item>
         <property name="text">
          <string>Frames</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Time</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>BBT</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="6" column="0" colspan="10">
    <widget class="QGroupBox" name="FadeInOutGroupBox">
     <property name="title">
      <string>Fade In/Out</string>
     </property>
     <layout class="QGridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="FadeInTextLabel">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Fade &amp;In:</string>
        </property>
        <property name="buddy">
         <cstring>FadeInLengthSpinBox</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qtractorTimeSpinBox" name="FadeInLengthSpinBox">
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Clip fade-in length</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QComboBox" name="FadeInTypeComboBox">
        <property name="toolTip">
         <string>Clip fade-in type</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3" rowspan="2">
       <spacer>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="FadeOutTextLabel">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Fade &amp;Out:</string>
        </property>
        <property name="buddy">
         <cstring>FadeOutLengthSpinBox</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qtractorTimeSpinBox" name="FadeOutLengthSpinBox">
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Clip fade-out length</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QComboBox" name="FadeOutTypeComboBox">
        <property name="toolTip">
         <string>Clip fade-out type</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="7" column="0" colspan="10">
    <widget class="QGroupBox" name="AudioClipGroupBox">
     <property name="title">
      <string>Audio</string>
     </property>
     <layout class="QGridLayout">
      <item row="0" column="2">
       <spacer>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>8</width>
          <height>8</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="TimeStretchTextLabel">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Ti&amp;me Stretch:</string>
        </property>
        <property name="buddy">
         <cstring>TimeStretchSpinBox</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="TimeStretchSpinBox">
        <property name="minimumSize">
         <size>
          <width>66</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Clip time-stretch percentage</string>
        </property>
        <property name="accelerated">
         <bool>true</bool>
        </property>
        <property name="suffix">
         <string> %</string>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="minimum">
         <double>10.0</double>
        </property>
        <property name="maximum">
         <double>1000.0</double>
        </property>
        <property name="singleStep">
         <double>0.1</double>
        </property>
        <property name="value">
         <double>100.0</double>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLabel" name="PitchShiftTextLabel">
        <property name="text">
         <string>Pitch S&amp;hift:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>PitchShiftSpinBox</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QDoubleSpinBox" name="PitchShiftSpinBox">
        <property name="minimumSize">
         <size>
          <width>66</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Clip pitch-shift in semitones</string>
        </property>
        <property name="accelerated">
         <bool>true</bool>
        </property>
        <property name="suffix">
         <string> semitones</string>
        </property>
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>-40.0</double>
        </property>
        <property name="maximum">
         <double>40.0</double>
        </property>
        <property name="singleStep">
         <double>0.01</double>
        </property>
        <property name="value">
         <double>0.0</double>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="4">
       <widget class="QCheckBox" name="WsolaTimeStretchCheckBox">
        <property name="font">
         <font>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <property name="toolTip">
         <string>Whether to use WSOLA time-stretching</string>
        </property>
        <property name="text">
         <string>&amp;WSOLA time-stretching</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="4">
       <widget class="QCheckBox" name="WsolaQuickSeekCheckBox">
        <property name="font">
         <font>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <property name="toolTip">
         <string>Whether to apply WSOLA quick seek time-stretching</string>
        </property>
        <property name="text">
         <string>WSOLA quic&amp;k seek</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="8" column="0" colspan="10">
    <layout class="QHBoxLayout">
     <item>
      <widget class="QCheckBox" name="ClipMuteCheckBox">
       <property name="text">
        <string>&amp;Mute</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="DialogButtonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="8" margin="8"/>
 <customwidgets>
  <customwidget>
   <class>qtractorTimeSpinBox</class>
   <extends>QSpinBox</extends>
   <header>qtractorSpinBox.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>ClipNameLineEdit</tabstop>
  <tabstop>FilenameComboBox</tabstop>
  <tabstop>FilenameToolButton</tabstop>
  <tabstop>TrackChannelSpinBox</tabstop>
  <tabstop>ClipStartSpinBox</tabstop>
  <tabstop>ClipOffsetSpinBox</tabstop>
  <tabstop>ClipLengthSpinBox</tabstop>
  <tabstop>ClipGainSpinBox</tabstop>
  <tabstop>ClipPanningSpinBox</tabstop>
  <tabstop>FormatComboBox</tabstop>
  <tabstop>FadeInLengthSpinBox</tabstop>
  <tabstop>FadeInTypeComboBox</tabstop>
  <tabstop>FadeOutLengthSpinBox</tabstop>
  <tabstop>FadeOutTypeComboBox</tabstop>
  <tabstop>TimeStretchSpinBox</tabstop>
  <tabstop>PitchShiftSpinBox</tabstop>
  <tabstop>WsolaTimeStretchCheckBox</tabstop>
  <tabstop>WsolaQuickSeekCheckBox</tabstop>
  <tabstop>ClipMuteCheckBox</tabstop>
  <tabstop>DialogButtonBox</tabstop>
 </tabstops>
 <resources>
  <include location="qtractor.qrc"/>
 </resources>
 <connections/>
</ui>
